app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -144,6 +144,12 @@ def perform_delete_in_cloud
       logger.warn("VMT with #{id_at_site} does not exist - continuing")
     end
 
+    # Returns the hourly cost for this template assuming a given VM flavor
+    def get_hourly_cost_for(flavor)
+      incarnation = flavor.flavor_os_families.find_by(os_family: appliance_type.os_family)
+      incarnation && incarnation.hourly_cost
+    end
```

Comment:
The is another possibility. While destroying `OsFamily` we can check if there is any template with this family assigned. If yes then refuse to destroy it.